### ISSUE TITLE
Import 'kogito-types.proto' not found

### ISSUE THREAD

**Issue by ChangDaoHang** [apache/incubator-kie-examples#769]
**Version**
```
<java.version>11</java.version>
<spring.version>2.5.2</version>
<kogito.version>1.8.0.Final</kogito.version>
```
**Spring POM**
```
...
        <dependency>
            <groupId>org.kie.kogito</groupId>
            <artifactId>kogito-api</artifactId>
        </dependency>
        <dependency>
            <groupId>org.kie.kogito</groupId>
            <artifactId>kogito-springboot-starter</artifactId>
        </dependency>

        <dependency>
            <groupId>org.kie.kogito</groupId>
            <artifactId>kogito-addons-persistence-mongodb</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-data-mongodb</artifactId>
        </dependency>
...
```
**[ERROR]**
When I use MongoDB for persistence,Kogito always throw the following exceptions:
```
...
Caused by: java.util.ServiceConfigurationError: org.kie.kogito.serialization.process.ObjectMarshallerStrategy: Provider org.kie.kogito.persistence.ProtostreamObjectMarshaller could not be instantiated
	at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:582) ~[na:na]
	at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:804) ~[na:na]
	at java.base/java.util.ServiceLoader$ProviderImpl.get(ServiceLoader.java:722) ~[na:na]
	at java.base/java.util.ServiceLoader$3.next(ServiceLoader.java:1395) ~[na:na]
	at org.kie.kogito.serialization.process.ProcessInstanceMarshallerService$Builder.withDefaultObjectMarshallerStrategies(ProcessInstanceMarshallerService.java:66) ~[process-serialization-protobuf-1.8.0.Final.jar:1.8.0.Final]
	at org.kie.kogito.mongodb.MongoDBProcessInstances.<init>(MongoDBProcessInstances.java:63) ~[kogito-addons-persistence-mongodb-1.8.0.Final.jar:1.8.0.Final]
	at org.kie.kogito.persistence.KogitoProcessInstancesFactory.createProcessInstances(KogitoProcessInstancesFactory.java:45) ~[kogito-addons-persistence-mongodb-1.8.0.Final.jar:na]
	at org.kie.kogito.persistence.KogitoProcessInstancesFactory.createProcessInstances(KogitoProcessInstancesFactory.java:29) ~[kogito-addons-persistence-mongodb-1.8.0.Final.jar:na]
	at org.kie.kogito.process.impl.AbstractProcess.configure(AbstractProcess.java:135) ~[jbpm-flow-1.8.0.Final.jar:1.8.0.Final]
	at org.kie.kogito.process.impl.AbstractProcess.activate(AbstractProcess.java:151) ~[jbpm-flow-1.8.0.Final.jar:1.8.0.Final]
	at com.example.demo.OrderItemsProcess.<init>(OrderItemsProcess.java:17) ~[classes/:na]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[na:na]
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:212) ~[spring-beans-5.3.8.jar:5.3.8]
	... 21 common frames omitted
Caused by: org.infinispan.protostream.DescriptorParserException: Import 'kogito-types.proto' not found
	at org.infinispan.protostream.descriptors.FileDescriptor.resolveImports(FileDescriptor.java:291) ~[protostream-4.4.1.Final.jar:4.4.1.Final]
	at org.infinispan.protostream.descriptors.FileDescriptor.resolveDependencies(FileDescriptor.java:229) ~[protostream-4.4.1.Final.jar:4.4.1.Final]
	at org.infinispan.protostream.descriptors.FileDescriptor.resolveDependencies(FileDescriptor.java:210) ~[protostream-4.4.1.Final.jar:4.4.1.Final]
	at org.infinispan.protostream.descriptors.ResolutionContext.resolve(ResolutionContext.java:57) ~[protostream-4.4.1.Final.jar:4.4.1.Final]
	at org.infinispan.protostream.impl.SerializationContextImpl.registerProtoFiles(SerializationContextImpl.java:127) ~[protostream-4.4.1.Final.jar:4.4.1.Final]
	at org.kie.kogito.persistence.ProtostreamObjectMarshaller.<init>(ProtostreamObjectMarshaller.java:22) ~[classes/:na]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[na:na]
	at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:780) ~[na:na]
	... 35 common frames omitted
```

**Comment by aymenmastouri** [comment]
I get also the same issue with MongoDB for persistence and Quarkus .
Same issue will be thrown by running the project kogito-examples/process-mongodb-persistence-quarkus/ (branche: stable)  in local dev Mode.

**Comment by ChangDaoHang** [comment]
This happens when I refer to the custom type as process data and use persistence in the local environment.

**Comment by ricardozanini** [comment]
cc @cristianonicolai @xieshenzh can you guys please take a look?

**Comment by xieshenzh** [comment]
It seems this issue is related to https://issues.redhat.com/browse/KOGITO-5320. @uteegozi is working on it.

**Comment by ricardozanini** [comment]
Many thanks, @xieshenzh!

@ChangDaoHang @Aymastoum you guys can follow the issue in the opened JIRA. Please let us know if we can help you with anything else.

**Comment by ChangDaoHang** [comment]
I now have to remove persistence to keep the service running properly. Although the index service caches data to the database, the application service cannot persist, so my service must always be running, otherwise the current process instance will be lost. I uploaded the example I'm trying, which is the SpringBoot project: [kogitoDemo](https://github.com/ChangDaoHang/kogitoDemo)

**Comment by ChangDaoHang** [comment]
@ricardozanini @xieshenzh @Aymastou I finally found the reason, which is something like this, when running locally, it is referenced in the generated "application-types.proto" file "Kogito-types. proto", but the path says" import "kogito-types.proto";"  In fact, I can't find the file because it is in the "META-INF" folder. I tried to change it to" import "meta-INF /kogito-types.proto";  Everything is normal!  
**before:**
![image](https://user-images.githubusercontent.com/82495367/127802484-d9afb7b6-5906-453c-8fb2-1bed14a2537c.png)

**after:**
![image](https://user-images.githubusercontent.com/82495367/127802508-ee751e52-adc7-46d5-a152-2fa92e8ba5f9.png)

Maybe Kogito can optimize this place。

**Comment by ricardozanini** [comment]
@cristianonicolai ^

**Comment by cristianonicolai** [comment]
@ChangDaoHang @Aymastou I checked both examples process-mongodb-persistence-springboot and process-mongodb-persistence-quarkus, on the current stable branch ( version 1.9.0 ) and both seem to work fine to start and initiate a new process instance using dev mode, spring-boot:run and the generated jars. Im not sure Im missing something or maybe its been fixed. Would you mind double checking and maybe giving us more details? Its just strange to have to change the folder path as both proto files are in the same META-INF folder.

**Comment by ChangDaoHang** [comment]
@cristianonicolai When I debug to the position shown in the third image, the dependency value is kogito-types.proto. Obviously, the dictionary value is not this, so fd is null and an exception is thrown.  If you want to reproduce this problem in your environment, you can pull the following code to reproduce it : https://github.com/ChangDaoHang/kogitoDemo
I was surprised too,maybe this is not kogito's problem, I am not familiar with Infinispan, so please take a look at what the problem is,tks.  
1.
![image](https://user-images.githubusercontent.com/82495367/128006789-0d9a120a-63b2-4304-9463-52ab82cc9bd2.png)
2.
![image](https://user-images.githubusercontent.com/82495367/128003345-32dd33dc-493f-4b56-a1c5-66e75d4aa17e.png)
3.
![16279878793984](https://user-images.githubusercontent.com/82495367/128003827-0080f9c2-5d22-4d70-80ee-88e1d8d53b52.png)
4.
![16279879444414](https://user-images.githubusercontent.com/82495367/128003850-d8a79e4d-df8a-4f61-91d3-382989dd4bca.png)
5.
![image](https://user-images.githubusercontent.com/82495367/128004650-7be18916-ca86-4d12-b7c0-cb7c61d2f796.png)

**Comment by JudeNiroshan** [comment]
Hi @ChangDaoHang, I'm investigating the errorneous scenario you described here. I am using kogito version 1.8.0.Final. 

I tried the [reproducible demo project](https://github.com/ChangDaoHang/kogitoDemo) which you shared. Apparently that project you have commented out the ```kogito-addons-persistence-mongodb``` dependency. Therefore the very first problem you reported on this issue that particular stack-trace will not appear. Because the class ```org.kie.kogito.persistence.ProtostreamObjectMarshaller``` will only be generated in the build time if you have that particular mongo dependency. 

I have added the mongo dependency and generated this class. Actually it has the correct protobuf file paths **including the META-INF** in the filepath.

Check the constructor of ```ProtostreamObjectMarshaller``` generated class. _(You need to have the ```kogito-addons-persistence-mongodb``` dependency)_

````
this.context.registerProtoFiles((new FileDescriptorSource())
.addProtoFile("kogito-types.proto", this.getClass().getClassLoader().getResourceAsStream("META-INF/kogito-types.proto"))
.addProtoFile("application-types.proto", this.getClass().getClassLoader().getResourceAsStream("META-INF/application-types.proto")));
````

However, I don't understand in your last screenshots(moving from 1 to 2), how did you end up with `FileDescriptor` class inside Infinispan? From which method does it delegate to Infinispan `resolveDependencies`? 

PS: I'm new to Kogito framework

**Comment by ChangDaoHang** [comment]
@JudeNiroshan Thank you for your information. According to your information,I have a deeper understanding of the problem. The path used when registering proto is "META-INF/kogito-types.proto", so this problem occurs.  Here's the thing: Kogito uses "META-INF/kogito-types.proto" to register, but the path referenced in file  "application-types.proto"  is "kogito-types.proto", which gives rise to the phenomenon I described above.  Now you can check the constructor of `ProtostreamObjectMarshaller` and enter the function `this.context.registerProtoFiles` , `SerializationContextImpl` is the implements `SerializationContext` , in `SerializationContextImpl` , you can see where the `resolutionContext.resolve();` is my entry point from above screenshots.  

ProtostreamObjectMarshaller:
```java
public class ProtostreamObjectMarshaller implements ObjectMarshallerStrategy {
    private static final String NAMESPACE = "org.kie.kogito/";
    private SerializationContext context = new SerializationContextImpl(Configuration.builder().build());

    public ProtostreamObjectMarshaller() {
        try {
            this.context.registerProtoFiles((new FileDescriptorSource()).addProtoFile("META-INF/kogito-types.proto", this.getClass().getClassLoader().getResourceAsStream("META-INF/kogito-types.proto")).addProtoFile("META-INF/application-types.proto", this.getClass().getClassLoader().getResourceAsStream("META-INF/application-types.proto")));
            this.context.registerMarshaller(new AddressMessageMarshaller());
            this.context.registerMarshaller(new TravellerMessageMarshaller());
            this.context.registerMarshaller(new StringProtostreamBaseMarshaller());
            this.context.registerMarshaller(new BooleanProtostreamBaseMarshaller());
            this.context.registerMarshaller(new DateProtostreamBaseMarshaller());
            this.context.registerMarshaller(new DoubleProtostreamBaseMarshaller());
            this.context.registerMarshaller(new FloatProtostreamBaseMarshaller());
            this.context.registerMarshaller(new IntegerProtostreamBaseMarshaller());
            this.context.registerMarshaller(new LongProtostreamBaseMarshaller());
        } catch (IOException var2) {
        }

    }
    ... ...
}
```
SerializationContextImpl:
```java
public void registerProtoFiles(FileDescriptorSource source) throws DescriptorParserException {
        if (log.isDebugEnabled()) {
            log.debugf("Registering proto files : %s", source.getFiles().keySet());
        }

        Map<String, FileDescriptor> fileDescriptorMap = this.parser.parse(source);
        this.dWriteLock.lock();

        try {
            Iterator var3 = fileDescriptorMap.keySet().iterator();

            while(var3.hasNext()) {
                String fileName = (String)var3.next();
                FileDescriptor oldFileDescriptor = (FileDescriptor)this.fileDescriptors.get(fileName);
                if (oldFileDescriptor != null) {
                    this.unregisterFileDescriptorTypes(oldFileDescriptor);
                }
            }

            this.fileDescriptors.putAll(fileDescriptorMap);
            ResolutionContext resolutionContext = new ResolutionContext(source.getProgressCallback(), this.fileDescriptors, this.genericDescriptors, this.typeIds, this.enumValueDescriptors);
            resolutionContext.resolve();
        } finally {
            this.dWriteLock.unlock();
        }

    }
```
PS: I'm new to Kogito framework too~ 😄 
PS2: Following your reminder, I updated the demo code.

**Comment by ricardozanini** [comment]
@cristianonicolai cc

**Comment by JudeNiroshan** [comment]
Hi @ChangDaoHang, Thanks for explaining the execute path. I would like to know, are you still able to reproduce the problem in the shared demo project? Because I tried the these attempts and I'm not able to see any problem. 

1. `mvn spring-boot:run`
2. `java -jar target/demo.jar`
3. running directly the main application class through IDEA

If you can still reproduce the bug with the shared demo project, it would be easier if you can share a docker file in that same repo which can reproduce the bug.

Let me briefly summarize execution flow. When we start this example application, we have added persistence add-on as dependency. Therefore, kogito maven plugin will identify it and generate persistence package with relevant java classes. ```ProtostreamObjectMarshaller``` is such a java class among few other. This class is being loaded at the startup of the application. It is being loaded through Java SPI mechanism. In the constructor our generated ```ProtostreamObjectMarshaller``` class populating a java hashmap which is used later during verifying the `.proto` file content. That's where it calls `.m2/repository/org/infinispan/protostream/protostream/4.4.1.Final/protostream-4.4.1.Final.jar!/org/infinispan/protostream/descriptors/FileDescriptor.class#resolveDependencies`. 

HashMap contains <String, FileDescriptor> which we registered in the constructor of ```ProtostreamObjectMarshaller```. As you can see, for the key we are simply using the filename **without** ```META-INF/```. Therefore, the `import "kogito-types.proto";` statement in the _application-types.proto_ should resolve the dependency. 

> The path used when registering proto is "META-INF/kogito-types.proto", so this problem occurs.

When registering we put the key string without the "META-INF/" part. It includes the "META-INF/" only to `this.getClass().getClassLoader().getResourceAsStream("META-INF/kogito-types.proto")` method which in this case, actual file content. 

I am not able to reproduce the error. Can you provide more insights of your reproducible setup?

**Comment by ChangDaoHang** [comment]
@JudeNiroshan The `ProtostreamObjectMarshaller` file is generated by the Kogito plugin and has one line of code in the constructor:
```java
... ...
this.context.registerProtoFiles((new FileDescriptorSource()).addProtoFile("META-INF/kogito-types.proto", this.getClass().getClassLoader().getResourceAsStream("META-INF/kogito-types.proto")).addProtoFile("META-INF/application-types.proto", this.getClass().getClassLoader().getResourceAsStream("META-INF/application-types.proto")));
...  ...
```
It was obviously registered using "META-INF/kogito-types.proto" as the map key,so I don't think what you're describing matches what I've seen with my own eyes.  I don't know why you can't reproduce this problem, but I think the demo is a pretty simple example.  So I'm sorry, I don't have more time to spend on this issue, thank you!

**Comment by cristianonicolai** [comment]
@ChangDaoHang I tried your demo and I'm also not able to reproduce the issue, managed to get the process working from end-to-end. Basically, I downloaded your project, updated the settings for my local mongo and kafka instances and run a `mvn clean install`. Proto files are generated correctly under `target/classes/META-INF`. To run the application I then executed `mvn clean package spring-boot:run`. From there I sent the curl commands as shown in [here](https://github.com/kiegroup/kogito-examples/tree/stable/process-mongodb-persistence-springboot#submit-a-deal). Perhaps this is due to some environment specific issue. Would be very helpful if you share more details, like jvm, maven version, etc so we can manage to reproduce this issue on our end.

**Comment by JudeNiroshan** [comment]
@ChangDaoHang 

ohh, that's the problem. See, the generated constructor for your `ProtostreamObjectMarshaller` is different. This is what I see on my generated `ProtostreamObjectMarshaller` file. 

```
public ProtostreamObjectMarshaller() {
        try {
            this.context.registerProtoFiles((new FileDescriptorSource()).addProtoFile("kogito-types.proto", this.getClass().getClassLoader().getResourceAsStream("META-INF/kogito-types.proto")).addProtoFile("application-types.proto", this.getClass().getClassLoader().getResourceAsStream("META-INF/application-types.proto")));
            this.context.registerMarshaller(new AddressMessageMarshaller());
            this.context.registerMarshaller(new TravellerMessageMarshaller());
            this.context.registerMarshaller(new StringProtostreamBaseMarshaller());
            this.context.registerMarshaller(new BooleanProtostreamBaseMarshaller());
            this.context.registerMarshaller(new DateProtostreamBaseMarshaller());
            this.context.registerMarshaller(new DoubleProtostreamBaseMarshaller());
            this.context.registerMarshaller(new FloatProtostreamBaseMarshaller());
            this.context.registerMarshaller(new IntegerProtostreamBaseMarshaller());
            this.context.registerMarshaller(new LongProtostreamBaseMarshaller());
        } catch (IOException var2) {
        }

    }
```
Note the first parameter value for addProtoFile method. In your generated class it has added `META-INF/` to the first parameter file name. This causes can cause the problem you are mentioning in this thread. Can you share the following runtime versions that you are using when you compiling this program?

- `java -version`
- `mvn -version`

I'm using the following versions:
- openjdk version "11.0.10"
- Apache Maven 3.8.1

**Comment by JudeNiroshan** [comment]
@ChangDaoHang @Aymastou This is most likely something related with kogito-runtime dependency version mismatch. When you run `mvn clean package`, can you check which version appears in the phase of model class generation? like below you can see my version is **`kogito-maven-plugin:1.8.0.Final:process-model-classes`**

```
[INFO] --- kogito-maven-plugin:1.8.0.Final:process-model-classes (default-process-model-classes) @ demo ---
[INFO] Performed addonsConfig discovery, found: AddonsConfig{usePersistence=true, useTracing=false, useMonitoring=false, usePrometheusMonitoring=false, useCloudEvents=false, useExplainability=false, useProcessSVG=true}
[INFO] Generating: org/kie/kogito/persistence/LongProtostreamBaseMarshaller.class
[INFO] Generating: org/kie/kogito/persistence/KogitoProcessInstancesFactoryImpl.class
[INFO] Generating: org/kie/kogito/persistence/ProtostreamObjectMarshaller.class
[INFO] Generating: org/kie/kogito/app/AddressMessageMarshaller.class
```

Removing the `META-INF/` in the `ProtostreamObjectMarshaller` constructor is being creating according to following logic. As per this, it should only have the proto file name _(1st parameter to `addProtoFile` method)_ without the `META-INF/`.

https://github.com/kiegroup/kogito-runtimes/blob/0d6e72d46ebea0792fad7f5a83734acf42d0c29d/kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/persistence/PersistenceGenerator.java#L328

**Comment by ChangDaoHang** [comment]
@JudeNiroshan @cristianonicolai 
To make it more intuitive, I made a simple recording. Please take a few minutes to watch it .
operation playback:
![动画2](https://user-images.githubusercontent.com/82495367/128801606-55176abf-2c48-45f9-9133-c727333bec05.gif)

java -version:
```
java version "11.0.11" 2021-04-20 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.11+9-LTS-194)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.11+9-LTS-194, mixed mode)
```
mvn -version：
```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: C:\Program Files\JetBrains\IntelliJ IDEA 2021.1.1\plugins\maven\lib\maven3\bin\..
Java version: 11.0.11, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk-11.0.11
Default locale: zh_CN, platform encoding: GBK
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
```
package log:
```sh
"C:\Program Files\Java\jdk-11.0.11\bin\java.exe" -Dmaven.multiModuleProjectDirectory=D:\0001-Work\01-project\01-phoenix\code\example\demo "-Dmaven.home=C:\Program Files\JetBrains\IntelliJ IDEA 2021.1.1\plugins\maven\lib\maven3" "-Dclassworlds.conf=C:\Program Files\JetBrains\IntelliJ IDEA 2021.1.1\plugins\maven\lib\maven3\bin\m2.conf" "-Dmaven.ext.class.path=C:\Program Files\JetBrains\IntelliJ IDEA 2021.1.1\plugins\maven\lib\maven-event-listener.jar" "-javaagent:C:\Program Files\JetBrains\IntelliJ IDEA 2021.1.1\lib\idea_rt.jar=21221:C:\Program Files\JetBrains\IntelliJ IDEA 2021.1.1\bin" -Dfile.encoding=UTF-8 -classpath "C:\Program Files\JetBrains\IntelliJ IDEA 2021.1.1\plugins\maven\lib\maven3\boot\plexus-classworlds-2.6.0.jar;C:\Program Files\JetBrains\IntelliJ IDEA 2021.1.1\plugins\maven\lib\maven3\boot\plexus-classworlds.license" org.codehaus.classworlds.Launcher -Didea.version=2021.2 --update-snapshots -s C:\Users\it1dzs\.m2\settings.xml -DskipTests=true package
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------------------< com.example:demo >--------------------------
[INFO] Building demo 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:3.2.0:resources (default-resources) @ demo ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] Copying 1 resource
[INFO] Copying 5 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ demo ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 5 source files to D:\0001-Work\01-project\01-phoenix\code\example\demo\target\classes
[INFO] 
[INFO] --- kogito-maven-plugin:1.8.0.Final:generateModel (default-generateModel) @ demo ---
[INFO] Performed addonsConfig discovery, found: AddonsConfig{usePersistence=true, useTracing=false, useMonitoring=false, usePrometheusMonitoring=false, useCloudEvents=false, useExplainability=false, useProcessSVG=true}
[INFO] Generator discovery performed, found [openapispecs, processes, rules, decisions, predictions]
[INFO] Generating: org/acme/deals/Dealreviews_2_TaskInput.java
[INFO] Generating: org/acme/deals/DealsProcess.java
[INFO] Generating: org/acme/deals/DealsModelOutput.java
[INFO] Generating: org/kie/kogito/app/ProcessServiceProducer.java
[INFO] Generating: org/acme/deals/Dealreviews_2_TaskOutput.java
[INFO] Generating: org/acme/deals/DealreviewsProcessInstance.java
[INFO] Generating: org/acme/deals/DealreviewsModelOutput.java
[INFO] Generating: org/acme/deals/DealsModel.java
[INFO] Generating: org/kie/kogito/app/CloudEventMetaFactory.java
[INFO] Generating: org/acme/deals/DealsModelInput.java
[INFO] Generating: org/acme/deals/DealsResource.java
[INFO] Generating: org/acme/deals/DealreviewsModel.java
[INFO] Generating: org/kie/kogito/app/EventMarshallerProducer.java
[INFO] Generating: org/acme/deals/Dealreviews_2_TaskModel.java
[INFO] Generating: org/acme/deals/DealsProcessInstance.java
[INFO] Generating: org/acme/deals/DealreviewsProcess.java
[INFO] Generating: org/acme/deals/Dealreviews_TaskModelFactory.java
[INFO] Generating: org/acme/deals/DealreviewsModelInput.java
[INFO] Generating: org/acme/deals/Deals_TaskModelFactory.java
[INFO] Generating: org/acme/deals/DealreviewsResource.java
[INFO] Generating: org/drools/project/model/SessionRuleUnit_defaultStatelessKieSession.java
[INFO] Generating: org/drools/project/model/SessionRuleUnit_defaultKieSession.java
[INFO] Generating: org/drools/project/model/ProjectModel.java
[INFO] Generating: org/drools/project/model/ProjectRuntime.java
[INFO] Generating: org/kie/kogito/app/Application.java
[INFO] Generating: org/kie/kogito/app/Processes.java
[INFO] Generating: org/kie/kogito/app/RuleUnits.java
[INFO] Generating: org/kie/kogito/app/DecisionModels.java
[INFO] Generating: org/kie/kogito/app/PredictionModels.java
[INFO] Generating: org/kie/kogito/app/ConfigBean.java
[INFO] Generating: org/kie/kogito/app/ProcessConfig.java
[INFO] Generating: org/kie/kogito/app/RuleConfig.java
[INFO] Generating: org/kie/kogito/app/DecisionConfig.java
[INFO] Generating: org/kie/kogito/app/PredictionConfig.java
[INFO] Generating: org/kie/kogito/app/ApplicationConfig.java
[INFO] Generating: org/kie/kogito/app/GlobalObjectMapper.java
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile-1) @ demo ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 41 source files to D:\0001-Work\01-project\01-phoenix\code\example\demo\target\classes
[INFO] 
[INFO] --- kogito-maven-plugin:1.8.0.Final:process-model-classes (default-process-model-classes) @ demo ---
[INFO] Performed addonsConfig discovery, found: AddonsConfig{usePersistence=true, useTracing=false, useMonitoring=false, usePrometheusMonitoring=false, useCloudEvents=false, useExplainability=false, useProcessSVG=true}
[INFO] Generating: org/kie/kogito/persistence/LongProtostreamBaseMarshaller.class
[INFO] Generating: org/kie/kogito/persistence/KogitoProcessInstancesFactoryImpl.class
[INFO] Generating: org/kie/kogito/persistence/ProtostreamObjectMarshaller.class
[INFO] Generating: org/kie/kogito/app/AddressMessageMarshaller.class
[INFO] Generating: org/kie/kogito/app/TravellerMessageMarshaller.class
[INFO] Generating: org/kie/kogito/persistence/DateProtostreamBaseMarshaller.class
[INFO] Generating: org/kie/kogito/persistence/StringProtostreamBaseMarshaller.class
[INFO] Generating: org/kie/kogito/persistence/FloatProtostreamBaseMarshaller.class
[INFO] Generating: org/kie/kogito/mongodb/transaction/MongoDBTransactionManagerImpl.class
[INFO] Generating: org/kie/kogito/persistence/BooleanProtostreamBaseMarshaller.class
[INFO] Generating: org/kie/kogito/persistence/DoubleProtostreamBaseMarshaller.class
[INFO] Generating: org/kie/kogito/persistence/IntegerProtostreamBaseMarshaller.class
[INFO] Generating: META-INF/resources/persistence/protobuf/dealreviews.proto
[INFO] Generating: META-INF/resources/persistence/protobuf/deals.proto
[INFO] Generating: META-INF/resources/persistence/protobuf/list.json
[INFO] Generating: META-INF/kogito-types.proto
[INFO] Generating: META-INF/application-types.proto
[INFO] Generating: META-INF/services/org.kie.kogito.serialization.process.ObjectMarshallerStrategy
[INFO] Generating: META-INF\jsonSchema\dealreviews_review.json
[INFO] 
[INFO] --- maven-resources-plugin:3.2.0:testResources (default-testResources) @ demo ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] skip non existing resourceDirectory D:\0001-Work\01-project\01-phoenix\code\example\demo\src\test\resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ demo ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 1 source file to D:\0001-Work\01-project\01-phoenix\code\example\demo\target\test-classes
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.2:test (default-test) @ demo ---
[INFO] Tests are skipped.
[INFO] 
[INFO] --- maven-jar-plugin:3.2.0:jar (default-jar) @ demo ---
[INFO] Building jar: D:\0001-Work\01-project\01-phoenix\code\example\demo\target\demo.jar
[INFO] 
[INFO] --- spring-boot-maven-plugin:2.5.2:repackage (repackage) @ demo ---
[INFO] Replacing main artifact with repackaged archive
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  49.743 s
[INFO] Finished at: 2021-08-10T09:35:22+08:00
[INFO] ------------------------------------------------------------------------

Process finished with exit code 0

```

`ProtostreamObjectMarshaller`:
![image](https://user-images.githubusercontent.com/82495367/128795167-babab918-f78d-4efe-91ae-7827ba93f682.png)

**Comment by cristianonicolai** [comment]
@ChangDaoHang thanks for the details, we managed to reproduce it, here is JIRA https://issues.redhat.com/browse/KOGITO-5687.

**Comment by fjtirado** [comment]
Here is the PR https://github.com/kiegroup/kogito-runtimes/pull/1523

**Comment by ChangDaoHang** [comment]
It's really a happy news~ Thanks for all your help, comrades. @JudeNiroshan @cristianonicolai @ricardozanini @fjtirado @Aymastou

**Comment by cristianonicolai** [comment]
@ChangDaoHang thank you for your patience and help with all the details

**Comment by ricardozanini** [comment]
> It's really a happy news~ Thanks for all your help, comrades. @JudeNiroshan @cristianonicolai @ricardozanini @fjtirado @Aymastou

The fix should be ready for 1.11.x.  @cristianonicolai do we have a window to include this PR on 1.10.x?

**Comment by cristianonicolai** [comment]
@ricardozanini unfortunately, build was in progress already when this was merged

**Comment by dfborba** [comment]
Is anyone still having this issue? I updated to version 1.11.1.Final, using last version of stable repository and still can't test process-mongodb-persistence-quarkus example. I cleaned everything in my environment.

Is something that I'm missing since it supposed to be fixes in the version 1.11.x and I'm running directly from example source?
 
mvn --version
```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: D:\tools\apache-maven-3.6.3\bin\..
Java version: 11.0.10, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk-11.0.10
Default locale: en_US, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
```

java --version
```
java 11.0.10 2021-01-19 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.10+8-LTS-162)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.10+8-LTS-162, mixed mode)
```

**Comment by cristianonicolai** [comment]
@dfborba should now be fixed by https://github.com/kiegroup/kogito-runtimes/pull/1647. That will land on 1.12
